pgpainless-core/src/test/java/org/pgpainless/key/modification/AddUserIdTest.java

```diff
@@ -84,4 +89,53 @@ public void deleteUserId_noSuchElementExceptionForMissingKey() throws IOExceptio
         assertThrows(NoSuchElementException.class, () -> PGPainless.modifyKeyRing(secretKeys)
                 .deleteUserId(0L, TestKeys.CRYPTIE_UID, new UnprotectedKeysProtector()));
     }
+
+    @Test
+    public void deleteExistingAndAddNewUserIdToExistingKeyRing()
+            throws InvalidAlgorithmParameterException, NoSuchAlgorithmException, PGPException, IOException {
+        final String ARMORED_PRIVATE_KEY =
```

Comment:
Could you change the test key to have two userIDs initially?
Since a PGP key must have at least one userID at all times, I plan to prevent the user from deleting the last userId (#69).
So if you add another userID I could merge this PR without having to change it later :)

Reply:
No, Paul, we'd like to use exactly this key. And I don't think it's correct to restrict adding/deleting user ids during key editing. Instead, I think `done()` should check that key finally has at least one user id and throw exception if this post-condition isn't met.

Reply:
Maybe it is worth to introduce editor option like `editor.allowDeleteLastUserId(boolean allow)` and still then always check post-condition in the `done()`.

Reply:
Sounds like a good alternative ;)